### ISSUE TITLE
No underline for Ale/Coc.nvim

### ISSUE THREAD

**Issue by bytesizedpcs** [sainnhe/gruvbox-material#29]
I'm not quite sure what is doing it, but for linting errors rendered by Coc.nvim, there's no underline.

There's underline available for other themes, such as Ayu.

**Comment by sainnhe** [comment]
What's the output of this command:

```vim
:hi ALEError
```

**Comment by bytesizedpcs** [comment]
```
:hi ALEError
ALEError    xxx cterm=underline gui=undercurl guisp=#ea6962
```

**Comment by sainnhe** [comment]
That is really strange.

Coc uses `CocErrorHighlight` to highlight error text, and ALE uses `ALEError` to highlight error text.

In this color scheme, `CocErrorHighlight` is linked to `ALEError`, and `ALEError is highlighted as underline/undercurl.

Let's do a very simple experiment:

```vim
:hi! link String ALEError
```

This command will link `String` to `ALEError`, this would theoretically make the strings underlined, like this

![demo1](https://user-images.githubusercontent.com/37491630/76674986-004faf00-65f0-11ea-8ce5-a8b5914f4a18.png)

You can try this command and see what will happen.

If the strings are not underlined, try this command and see what will happen:

```vim
:hi String gui=underline cterm=underline
```

Btw, could you provide the following information?

- The client you are using(Vim, GVim, Neovim, etc.)
- The version of your vim client.
- If you're using vim/nvim in terminal, please provide the info of your terminal emulator.
- Output of `:echo has('termguicolors')` and `:echo has('gui_running')`

**Comment by bytesizedpcs** [comment]
So, spooling up my terminal to try your experiments, and I opened vim and it had the correct underlines.

When I started Tmux and opened the same file, it didn't have any underlines.

`:hi! link String ALEError` command changed all the strings from green to a tan (paren color)

`:hi String gui=underline cterm=underline` command changed all the strings from tan to green and underlined

Neovim
![image](https://user-images.githubusercontent.com/7506717/76675518-a4097280-6590-11ea-8ec7-0808de78dd0f.png)
Terminal: Tilix, running in Tmux (`tmux -2` when running)
`:echo has('termguicolors') = 1`
`:echo has('gui_running') = 0`

**Comment by sainnhe** [comment]
Looks like some terminal emulators don't support `undercurl`. I pushed a commit b8efb73, could you update this color scheme and get it a try?

**Comment by bytesizedpcs** [comment]
The underline works now while in Tmux, but the undercurl is gone while not in Tmux.

**Comment by sainnhe** [comment]
Should be fixed via 285ba13.

**Comment by bytesizedpcs** [comment]
Looks like it's working now!
